doc/modules/lmnn.rst

```diff
@@ -0,0 +1,161 @@
+.. _lmnn:
+
+=============================
+Large Margin Nearest Neighbor
+=============================
+
+.. sectionauthor:: John Chiotellis <johnyc.code@gmail.com>
+
+.. currentmodule:: sklearn.neighbors
+
+Large Margin Nearest Neighbor (:class:`lmnn.LargeMarginNearestNeighbor`) is
+a distance metric learning algorithm.
+
+This transformer aims to learn a Mahalanobis metric that improves the
+accuracy of k-nearest neighbor classification compared to the standard
+Euclidean metric.
+
+As in support vector machines (SVMs), this method attempts to maximize the
+margin between data samples that belong to different classes, therefore the
+name of the method. Instead of a set of hyper-planes in a high or infinite
+dimensional space, LMNN constructs a set of hyper-spheres centered at each
+training point that are supposed to contain only samples that share the same
+label as the central point.
+
+In the beginning of the algorithm each training sample fixes :math:`k`
+target neighbors, namely the :math:`k` nearest training samples that share
+the same label according to some initial metric - most commonly the Euclidean.
+
+The first goal of the algorithm is to find a metric such that the distances of
+each sample to its target neighbors are minimized. The second goal of the
+algorithm is to create a large margin between the farthest target neighbor
```

Comment:
again this is not completely clear. maybe something like "the second goal of the algorithm is to make each point closer to its target neighbors than to differently-labeled points by a large margin"?